test/src/main/java/io/strimzi/test/BaseITST.java

```diff
@@ -0,0 +1,188 @@
+                !file.getName().matches(".*(Binding|Deployment)-.*")
+        ).collect(Collectors.toMap(file -> file, f -> TestUtils.getContent(f, TestUtils::toYamlString), (x, y) -> x, LinkedHashMap::new));
+        for (Map.Entry<File, String> entry : clusterOperatorMap.entrySet()) {
+            LOGGER.info("Applying configuration file: {}", entry.getKey());
+            KUBE_CLIENT.clientWithAdmin().applyContent(entry.getValue());
+        }
+        TimeMeasuringSystem.stopOperation(Operation.CO_CREATION);
+    }
+
+    /**
+     * Delete ServiceAccount, Roles and CRDs from kubernetes cluster.
+     */
+    private void deleteClusterOperatorInstallFiles() {
+        TimeMeasuringSystem.setTestName(testClass, testClass);
+        TimeMeasuringSystem.startOperation(Operation.CO_DELETION);
+        for (Map.Entry<File, String> entry : clusterOperatorMap.entrySet()) {
+            LOGGER.info("Removing configuration file: {}", entry.getKey());
+            KUBE_CLIENT.clientWithAdmin().deleteContent(entry.getValue());
+        }
+        TimeMeasuringSystem.stopOperation(Operation.CO_DELETION);
+    }
+
+    /**
+     * Create namespaces for test resources.
+     * @param useNamespace namespace which will be used as default by kubernetes client
+     * @param namespaces list of namespaces which will be created
+     */
+    private void createNamespaces(String useNamespace, List<String> namespaces) {
+        deploymentNamespaces = namespaces;
+        for (String namespace: namespaces) {
+            LOGGER.info("Creating namespace: {}", namespace);
+            KUBE_CLIENT.createNamespace(namespace);
+            KUBE_CLIENT.waitForResourceCreation("Namespace", namespace);
+            LOGGER.info("Namespace {} created", namespace);
+        }
+        LOGGER.info("Using namespace {}", useNamespace);
+        KUBE_CLIENT.namespace(useNamespace);
+    }
+
+    /**
+     * Create namespace for test resources.
+     * @param useNamespace namespace which will be created and used as default by kubernetes client
+     */
+    protected void createNamespace(String useNamespace) {
+        createNamespaces(useNamespace, Collections.singletonList(useNamespace));
+    }
+
+    /**
+     * Delete all created namespaces.
+     */
+    protected void deleteNamespaces() {
+        for (String namespace: deploymentNamespaces) {
+            LOGGER.info("Deleting namespace: {}", namespace);
+            KUBE_CLIENT.deleteNamespace(namespace);
+            KUBE_CLIENT.waitForResourceDeletion("Namespace", namespace);
+            LOGGER.info("Namespace {} deleted", namespace);
+        }
+        LOGGER.info("Using namespace {}", CLUSTER.defaultNamespace());
+        KUBE_CLIENT.namespace(CLUSTER.defaultNamespace());
+    }
+
+    /**
+     * Apply custom resources for CO such as templates.
+     * @param resources list of paths to yaml files with resources specifications
```

Comment:
Mention that it's up to the caller to arrange deletion by calling `deleteCustomResources()`

Reply:
Or by calling `teardownEnvForOperator()` I suppose